src/brainfuck/Interpreter.java

```diff
@@ -27,15 +34,29 @@ public Interpreter(List<Instruction> instructions) {
 	}
 
 	/**
+	 * Set the Logger.
+	 *
+	 * @param log	Logger to use for this instance of the Interpreter.
+	 */
+	public void setLogger(Logger log) {
+		this.logger = log;
+	}
+
+	/**
 	 * Executes the instructions from the List and print memory content if the program terminated successfully.
 	 * Parses in reverse order when Machine::isReversed returns true.
 	 *
 	 * @param machine	Virtual machine which executes the instructions.
+	 * @throws IOException	if any attempt to write in the log file failed.
 	 */
-	public void run(Machine machine) {
+	public void run(Machine machine) throws IOException {
 		int i = 0;
+		int j = 0;
 		while (i >= 0 && i < instructions.size()) {
 			machine.executeOp(instructions.get(i));
+			if (logger != null) {
+				logger.write((++j)+" — exec "+i+": "+instructions.get(i).getName()+" on C"+machine.getLocation()+"\n"+machine.dumpMemory());
```

Comment:
Should still be done inside the `Logger` class in my opinion, calling a method `logStep()` (or something similar explicit enough) and passing it the necessary info to build the String.

Also store the result of the `instructions.get()` in a variable to avoid doing it two times.